python/cudf/cudf/io/parquet.py

```diff
@@ -558,6 +558,17 @@ def read_parquet(
                 "for full CPU-based filtering functionality."
             )
 
+    # Make sure we read in the columns needed for row-wise
+    # filtering after IO. This means that one or more columns
+    # will be dropped almost immediately after IO. However,
+    # we do NEED these columns for accurate filtering.
+    projection = None
+    if columns and filters:
+        filtered_columns = _filtered_columns(filters)
```

Comment:
At this point you've normalized the filters into `list[list[tuple]]` so `_filtered_columns` doesn't need to recurse to flatten things, and you can just do:
```
filtered_columns = set(v[0] for v in itertools.chain.from_iterable(filters))
```

Perhaps then:
```python
projected_columns = columns # this could be None so it's fine
if columns and filters:
    projected_columns = columns
    columns = sorted(set(v[0] for v in itertools.chain.from_iterable(filters)) | set(columns))
```
WDYT?

Reply:
Ah, good point about `filters` being normalized here. In that case, `itertools.chain` makes a lot of sense.

> `projected_columns = columns # this could be None so it's fine`

Interestingly, this line caused failures, and exposed a bug in my logic. `columns` can technically include fields that are used to set the index, and so we need to make sure `projected_columns` only includes current column names before the `getitem` operation at the end of this function.

Since we need extra logic to check the elements of  `projected_columns`, I decided it probably makes more sense for the `projected_columns`  default to be `None` (rather than `columns`).